src/Sylius/Component/Storage/CookieStorage.php

```diff
@@ -53,6 +64,12 @@ public function getData($key, $default = null)
     public function setData($key, $value)
     {
         $this->request->cookies->set($key, $value);
+
+        $this->eventDispatcher->addListener('kernel.response', function (FilterResponseEvent $event) use ($key, $value) {
```

Comment:
@pjedrzejewski Perhaps we should move this to CoreBundle?

Symfony 2.3 doesn't support PSR-7 anyway.